docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,339 @@
+
+**NOTE: Some routing terms are defined in the [glossary](./glossary.md) file, to more deeply describe how we will handle behaviors like priority and fairness**
+
+## Goals
+
+- Drive concensus on direction of LLM Instance Gateway Solution
+- Documentation of API decisions for posterity
+
+## Non-Goals
+
+- Hash out every implementation detail
+- Be a formal KEP
+
+## Proposal
+
+### Personas
+
+Before diving into the details of the API, decriptions of the personas will help shape the thought process of the API design.
+
+#### Inference Platform Admin
+
+The Inference Platform Admin creates and manages the infrastructure necessary to run LLM workloads. Including handling Ops for: 
+  - Hardware
+  - Model Server
+  - Base Model
+  - Resource Allocation for Workloads
+  - Gateway configuration
+  - etc
+
+#### LLM Use Case Owner
+
+An LLM Use Case Owner persona owns and manages 1 or many Generative AI Workloads (LLM focused *currently*). This includes:
+- Defining SLO
+- Managing fine-tunes
+  - LoRA Adapters
+  - System Prompts
+  - Prompt Cache
+  - etc.
+- Managing rollout of adapters
+
+### Axioms 
+
+The API design is based on these axioms:
+
+- Pools of shared compute should be *discrete* for scheduling to properly work
+- Pod-level scheduling should not be handled by a high-level gateway 
+- Simple use cases should be simple to define (or are implicitly defined via reasonable defaults)
+- This solution should be composable with other Gateway solutions and flexible to fit customer needs
+- The MVP will heavily assume requests are done using the OpenAI spec, but open to extension in the future
+- The Gateway should route in a way that does not generate a queue of requests at the model server level
+
+The [PoC](https://youtu.be/NUBZg_uqqXk?si=v681EeYdGUGEVqQQ&t=1458) was focused on lower-level scheduling. And the API follows that similar logic, which lead to the proposal of the **LLMServerPool**.
+
+### LLMServerPool
+
+The LLMServerPool at its core is a logical grouping of compute, expressed in the form of Pods (typically model servers), akin to a K8s Service. The LLMServerPool would deploy its own routing, and offer administrative configuration to the Platform Admin. 
+
+ It is expected for the LLMServerPool to:
+ - Enforce fair consumption of resources across competing use cases
+ - Efficiently route requests across shared compute (as displayed by the PoC)
+ 
+It is _not_ expected for the LLMServerPool to:
+ - Enforce any common set of adapters or base models are available on the Pods
+ - Manage Deployments of Pods within the Pool
+ - Manage Pod lifecycle of pods within the pool 
+
+Additionally, any Pod that seeks to join a LLMServerPool would need to support a protocol, defined by LLM Instance Gateway, to ensure the Pool has adequate information to intelligently route requests.
+
+### LLMService
+
+A LLMService allows the LLMServiceOwner to define:
```

Comment:
nit: did we define LLMServiceOwner somewhere? probably LLM Service Owner?

Reply:
yes, we are calling it "LLM Service Owner" not LLMServiceOwner, no spaces makes it look like an object/struct